src/extension.ts

```diff
@@ -1,176 +1,208 @@
-    return themeList;
-  }
+    const currentTheme = userSettings.get('workbench.colorTheme', '');
+    if (themeList.length === 1) {
+      vscode.window.showInformationMessage('Why only one theme ç_ç');
+      r(themeList);
+    }
 
-  return themeList.filter(theme => theme !== currentTheme);
+    r(themeList.filter(theme => theme !== currentTheme));
+
+  });
 }
 
-function addCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
-  const themeList = getThemeList(extensionConfig, userSettings);
+async function addCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
+  const themeList = await getThemeList(extensionConfig, userSettings);
 
   const currentThemeName: string | undefined = userSettings.get('workbench.colorTheme');
 
   if (typeof currentThemeName !== 'undefined') {
-    saveThemes(themeList.concat(currentThemeName).sort(), 'addedtheme', currentThemeName);
+    saveThemes(themeList.concat(currentThemeName).sort(), Messages.AddedTheme(currentThemeName));
   }
 }
 
-function removeCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
-  const themeList = getThemeList(extensionConfig, userSettings);
+async function removeCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
+  const themeList = await getThemeList(extensionConfig, userSettings);
 
   const currentThemeName: string | undefined = userSettings.get('workbench.colorTheme');
 
   if (typeof currentThemeName !== 'undefined') {
-    saveThemes(themeList.filter(th => th !== currentThemeName), 'removedtheme', currentThemeName);
+    saveThemes(themeList.filter(th => th !== currentThemeName), Messages.RemovedTheme(currentThemeName));
   }
 }
 
-export async function activate(context: vscode.ExtensionContext) {
-  const switchTheme = vscode.commands.registerCommand('randomThemeSwitcher.switchTheme', () => {
-    changeTheme();
-  });
-
-  context.subscriptions.push(switchTheme);
-
-  const copyThemesToSettings = vscode.commands.registerCommand('randomThemeSwitcher.copyInstalledThemes', () => {
-    const installedThemes = getInstalledThemes();
-
-    saveThemes(installedThemes, 'copyall');
-  });
-
-  context.subscriptions.push(copyThemesToSettings);
-
-  const addCurrentThemeToSettogs = vscode.commands.registerCommand('randomThemeSwitcher.addCurrentTheme', () => {
-    addCurrentTheme();
-  });
-
-  context.subscriptions.push(addCurrentThemeToSettogs);
-
-  const removeCurrentThemeToSettogs = vscode.commands.registerCommand('randomThemeSwitcher.removeCurrentTheme', () => {
-    removeCurrentTheme();
-  });
+let switchMode: SwitchModes = 'manual';
+let isSwitching = false;
+let themeList: string[] = new Array();
 
-  context.subscriptions.push(removeCurrentThemeToSettogs);
+export async function activate(context: vscode.ExtensionContext) {
 
-  const extensionConfig = getExtensionConfig();
+  let extensionConfig = getExtensionConfig();
```

Comment:
Come mai hai cambiato da `const` a `let`? non mi pare venga riassegnata la variabile (potrei sbagliarmi)

Reply:
Speravo di ridurre lo spazio unendo tutte le funzioni in un'unica `extensionConfig` risultante,
Purtroppo non ha funzionato e ho dovuto rinunciare, perchè mi ha dato problemi di sincronizzazione:
tipo nello SwitchTheme se gli passi la variabile non cambia più (penso che abbia a che fare con la variabile che resta indietro rispetto alla configurazione reale).
Così ho testato e funziona.